Add .Scratch.DeleteInMap

Unless we revive either https://github.com/gohugoio/hugo/issues/7232 or https://github.com/gohugoio/hugo/issues/5604 `Scratch` will remain the only efficient way to edit maps in Hugo.

## The problem
For now, the only thing missing is a way to remove a key from a map stored in a `Scratch` key.

([.Scratch.Delete](https://gohugo.io/functions/scratch/#delete) will remove a key from the the `Scratch` instance itself.

## Proposal

The method called `.DeleteInMap` to resonate with the familiar `.SetInMap`, would take two parameters: 
the key for the map, and the key to remove.

```
{{ $s := newScratch }}
{{ $s.Set "data" (dict
  "hello" "bonjour"
  "goodbye" "au revoir"
  "car" "voiture"
) }}

{{ $s.DeleteInMap "data" "car" }}
```

@bep I was having issues with a package called `io/fs`. I tried recloning the repo and running `go install` again but I still keep getting the following error

```
tpl\internal\go_templates\texttemplate\helper.go:11:2: package io/fs is not in GOROOT (C:\Program Files\Go\src\io\fs)
```

Running `go version` gives the following output:

```
go version go1.15.8 windows/amd64
```

Does hugo require a certain version of `go` for the build to be successful?

--- a/common/maps/scratch.go
+++ b/common/maps/scratch.go
@@ -129,6 +129,17 @@ func (c *Scratch) SetInMap(key string, mapKey string, value interface{}) string 
 	return ""
 }
 
+// DeleteInMap deletes a value to a map with the given key in the Node context.
+func (c *Scratch) DeleteInMap(key string, mapKey string) string {
+	c.mu.Lock()
+	_, found := c.values[key]
+	if found {
+		delete(c.values[key].(map[string]interface{}), mapKey)
+	}
+	c.mu.Unlock()
+	return ""
+}
+
 // GetSortedMapValues returns a sorted map previously filled with SetInMap.
 func (c *Scratch) GetSortedMapValues(key string) interface{} {
 	c.mu.RLock()

--- a/common/maps/scratch_test.go
+++ b/common/maps/scratch_test.go
@@ -188,6 +188,20 @@ func TestScratchSetInMap(t *testing.T) {
 	c.Assert(scratch.GetSortedMapValues("key"), qt.DeepEquals, []interface{}{0: "Abc (updated)", 1: "Def", 2: "Lux", 3: "Zyx"})
 }
 
+func TestScratchDeleteInMap(t *testing.T) {
+	t.Parallel()
+	c := qt.New(t)
+
+	scratch := NewScratch()
+	scratch.SetInMap("key", "lux", "Lux")
+	scratch.SetInMap("key", "abc", "Abc")
+	scratch.SetInMap("key", "zyx", "Zyx")
+	scratch.DeleteInMap("key", "abc")
+	scratch.SetInMap("key", "def", "Def")
+	scratch.DeleteInMap("key", "lmn") // Do nothing
+	c.Assert(scratch.GetSortedMapValues("key"), qt.DeepEquals, []interface{}{0: "Def", 1: "Lux", 2: "Zyx"})
+}
+
 func TestScratchGetSortedMapValues(t *testing.T) {
 	t.Parallel()
 	scratch := NewScratch()

--- a/docs/content/en/functions/scratch.md
+++ b/docs/content/en/functions/scratch.md
@@ -97,6 +97,18 @@ Takes a `key`, `mapKey` and `value` and add a map of `mapKey` and `value` to the
 {{ .Scratch.Get "greetings" }} > map[french:Bonjour english:Hello]
 ```
 
+#### .DeleteInMap
+Takes a `key` and `mapKey` and removes the map of `mapKey` from the given `key`.
+
+```go-html-template
+{{ .Scratch.SetInMap "greetings" "english" "Hello" }}
+{{ .Scratch.SetInMap "greetings" "french" "Bonjour" }}
+----
+{{ .Scratch.DeleteInMap "greetings" "english" }}
+----
+{{ .Scratch.Get "greetings" }} > map[french:Bonjour]
+```
+
 #### .GetSortedMapValues
 Returns array of values from `key` sorted by `mapKey`
 

